### ISSUE TITLE
Retry limit of 0 seizes

### ISSUE THREAD

**Issue by kriskowal** [uber/tchannel-python#154]
But retry limit of 1 works fine.

Steps to reproduce:

```python
yield tchannel.thrift(request, retry_limit=0)
print 'got here'
```

Expected outcome: "got here"
Actual outcome: nothin'

**Comment by breerly** [comment]
Yeah this is a bug... it should try 4 times though, not seize...

**Comment by breerly** [comment]
fixed in `0.16.2`.
